web/src/features/hfiCalculator/RowManager.ts

```diff
@@ -0,0 +1,249 @@
+import { FireCentre, PlanningArea } from 'api/hfiCalcAPI'
+import { StationDaily } from 'api/hfiCalculatorAPI'
+import { groupBy, isNull, isUndefined, range } from 'lodash'
+import { getDailiesByStationCode, getDailiesForArea } from 'features/hfiCalculator/util'
+import { dailyTableColumnLabels } from 'features/hfiCalculator/components/DailyViewTable'
+import { weeklyTableColumnLabels } from 'features/hfiCalculator/components/WeeklyViewTable'
+import {
+  calculateDailyMeanIntensities,
+  calculateMaxMeanIntensityGroup,
+  calculateMeanIntensity,
+  calculateMeanIntensityGroupLevel
+} from 'features/hfiCalculator/components/meanIntensity'
+import { calculatePrepLevel } from 'features/hfiCalculator/components/prepLevel'
+import { isValidGrassCure } from 'features/hfiCalculator/validation'
+import { NUM_WEEK_DAYS } from 'features/hfiCalculator/constants'
+
+// the number of decimal places to round to
+const DECIMAL_PLACES = 1
+
+const printGrassCurePercentage = (daily: StationDaily): string => {
+  if (!isUndefined(daily) && !isNull(daily.grass_cure_percentage)) {
+    return daily.grass_cure_percentage.toString()
+  } else {
+    return 'ND'
+  }
+}
+
+export class RowManager {
+  public static exportDailyRowsAsStrings = (
```

Comment:
I'm concerned about the disconnect; I was thinking you'd have something along these lines:
```
rows = makeTheRows()
// so that:
gimmeCsv(rows)
// and so that
<MyTable rows=rows/>
```
Why is there such a big different between the csv, and what's being displayed in the html?

Reply:
Remember near the start of this ticket (a very long time ago, I know) when I had a dev chat with you & Conor about this - initially I tried to just export the HTML table to a string but it wasn't working with Excel at all, so we all agreed that the most efficient approach would be to just manually construct the CSV string

Reply:
Oh yeah - I'm not saying HTML -> csv ; I'm saying 0) logic -> data 1) data -> html 2) data -> csv ; right now the pattern seems to be 0) logic -> blah 1) blah -> data -> html 2) blah -> other data -> csv) ;
It's a concern - I'm ok not addressing it right now, just feel it's going to come back an bite us.